File: Interview/add-strings.java
//  Add Strings

// Time O(M+N), Space O(N)
class Solution {
    public String addStrings(String num1, String num2) {
        int p1 = num1.length()-1, p2 = num2.length()-1, k = 0;
        StringBuffer sb = new StringBuffer();
        while (p1>=0 || p2>=0) {
            int t = (p1>=0 ? num1.charAt(p1--)-'0' : 0) + (p2>=0 ? num2.charAt(p2--)-'0' : 0) + k;
            sb.append(t%10);
            k = t / 10;
        }
        if (k == 1)
            sb.append(1);
        return sb.reverse().toString();
    }
}

File: Airbnb/add-strings.cc
//  Add Strings

// Time O(M+N), Space O(N)
class Solution {
public:
    string addStrings(string num1, string num2) {
        string res = "";
        int p1 = num1.size()-1, p2 = num2.size()-1;
        int carry = 0;
        while (p1>=0 || p2>=0) {
            int sum = (p1>=0 ? num1[p1]-'0' : 0) + (p2>=0 ? num2[p2]-'0' : 0) + carry;
            res = to_string(sum % 10) + res;
            carry = sum / 10;
            p1 --;
            p2 --;
        }
        if (carry != 0)
            res = to_string(carry) + res;
        return res;
    }
};